cmd/tailscaled/tailscaled.go

```diff
@@ -511,7 +511,13 @@ func getLocalBackend(ctx context.Context, logf logger.Logf, logID logid.PublicID
 			return ok
 		}
 		dialer.NetstackDialTCP = func(ctx context.Context, dst netip.AddrPort) (net.Conn, error) {
-			return ns.DialContextTCP(ctx, dst)
+			// Note: don't just return ns.DialContextTCP or we'll
+			// return an interface containing a nil pointer.
```

Comment:
The comments are good but it might be worth writing a little test to ensure there's no regressions of this behaviour.

Reply:
I tried that, but it's super annoying to test this since it's in the `main` package, and requires a running Tailscale backend to even get this far 😵‍💫